container-disc/src/test/java/com/yahoo/container/jdisc/metric/GarbageCollectionMetricsTest.java

```diff
@@ -0,0 +1,60 @@
+package com.yahoo.container.jdisc.metric;
+
+import com.yahoo.jdisc.Metric;
+import org.junit.Test;
+
+import java.lang.management.ManagementFactory;
+import java.time.Clock;
+import java.util.LinkedList;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author ollivir
+ */
+public class GarbageCollectionMetricsTest {
+    @Test
+    public void gc_metrics_reported_at_intervals() {
+        Clock clock = mock(Clock.class);
```

Comment:
In vespa:testutil there is a ManualClock class that probably could be used instead of mocking.